### ISSUE TITLE
SQL error in computer search when field "external link" is selected in the view (both in general view or personnal view)

### ISSUE THREAD

**Issue by redisded** [glpi-project/glpi#7763]
**Describe the bug**

Since upgrade in 9.5.1, in the computer page, when the field "external link" is selected to be shown in the results, the search page throw the following error  : 

`SQL Error "1370": execute command denied to user 'glpi'@'%' for routine 'glpi_links_c33a34eda48888ce7667b9068fa121e0.id' in query "SELECT DISTINCT `glpi_computers`.`id` AS id, 'my_user' AS currentuser, `glpi_computers`.`entities_id`, `glpi_computers`.`is_recursive`, `glpi_computers`.`name` AS `ITEM_Computer_1`, `glpi_computers`.`id` AS `ITEM_Computer_1_id`, `glpi_entities`.`completename` AS `ITEM_Computer_80`, `glpi_computers`.`serial` AS `ITEM_Computer_5`, `glpi_states`.`completename` AS `ITEM_Computer_31`, `glpi_computertypes`.`name` AS `ITEM_Computer_4`, `glpi_computermodels`.`name` AS `ITEM_Computer_40`, `glpi_computers`.`contact` AS `ITEM_Computer_7`, `glpi_manufacturers`.`name` AS `ITEM_Computer_23`, `glpi_operatingsystemversions_9719987b154aaf3b42c3db32aef59090`.`name` AS `ITEM_Computer_46`, GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`name`, '__NULL__'), '$#$',`glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`id`) ORDER BY `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`id` SEPARATOR '$$##$$') AS `ITEM_Computer_126`, `glpi_plugin_fusioninventory_inventorycomputercomputers`.`last_fusioninventory_update` AS `ITEM_Computer_5150`, `glpi_plugin_fusioninventory_inventorycomputercomputers`.`last_boot` AS `ITEM_Computer_5182`, IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_c33a34eda48888ce7667b9068fa121e0`.`id`, '__NULL__'), '$#$', `glpi_links_c33a34eda48888ce7667b9068fa121e0`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$') AS `ITEM_Computer_145_id`, IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_c33a34eda48888ce7667b9068fa121e0`.`link`, '__NULL__'), '$#$', `glpi_links_c33a34eda48888ce7667b9068fa121e0`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$') AS `ITEM_Computer_145_link`, IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_c33a34eda48888ce7667b9068fa121e0`.`name`, '__NULL__'), '$#$', `glpi_links_c33a34eda48888ce7667b9068fa121e0`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$') AS `ITEM_Computer_145_name`, IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_c33a34eda48888ce7667b9068fa121e0`.`data`, '__NULL__'), '$#$', `glpi_links_c33a34eda48888ce7667b9068fa121e0`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$') AS `ITEM_Computer_145_data`, IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_c33a34eda48888ce7667b9068fa121e0`.`open_window`, '__NULL__'), '$#$', `glpi_links_c33a34eda48888ce7667b9068fa121e0`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$') AS `ITEM_Computer_145_open_window` FROM `glpi_computers`LEFT JOIN `glpi_entities` ON (`glpi_computers`.`entities_id` = `glpi_entities`.`id` )LEFT JOIN `glpi_states` ON (`glpi_computers`.`states_id` = `glpi_states`.`id` )LEFT JOIN `glpi_computertypes` ON (`glpi_computers`.`computertypes_id` = `glpi_computertypes`.`id` )LEFT JOIN `glpi_computermodels` ON (`glpi_computers`.`computermodels_id` = `glpi_computermodels`.`id` )LEFT JOIN `glpi_manufacturers` ON (`glpi_computers`.`manufacturers_id` = `glpi_manufacturers`.`id` ) LEFT JOIN `glpi_items_operatingsystems` ON (`glpi_computers`.`id` = `glpi_items_operatingsystems`.`items_id` AND `glpi_items_operatingsystems`.`itemtype` = 'Computer' ) LEFT JOIN `glpi_operatingsystemversions` AS `glpi_operatingsystemversions_9719987b154aaf3b42c3db32aef59090` ON (`glpi_items_operatingsystems`.`operatingsystemversions_id` = `glpi_operatingsystemversions_9719987b154aaf3b42c3db32aef59090`.`id` ) LEFT JOIN `glpi_ipaddresses` AS `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540` ON (`glpi_computers`.`id` = `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`mainitems_id` AND `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`mainitemtype` = 'Computer' AND `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`is_deleted` = 0 ) LEFT JOIN `glpi_plugin_fusioninventory_inventorycomputercomputers` AS glpi_plugin_fusioninventory_inventorycomputercomputers ON (`glpi_computers`.`id` = `glpi_plugin_fusioninventory_inventorycomputercomputers`.`computers_id` ) LEFT JOIN `glpi_links_itemtypes` ON (`glpi_links_itemtypes`.`itemtype` = 'Computer' ) LEFT JOIN `glpi_links` AS `glpi_links_c33a34eda48888ce7667b9068fa121e0` ON (`glpi_links_itemtypes`.`links_id` = `glpi_links_c33a34eda48888ce7667b9068fa121e0`.`id` (`glpi_links_c33a34eda48888ce7667b9068fa121e0`.`entities_id` IN ('153')) ) WHERE `glpi_computers`.`is_deleted` = 0 AND `glpi_computers`.`is_template` = 0 AND ( `glpi_computers`.`entities_id` IN ('153') OR (`glpi_computers`.`is_recursive`='1' AND `glpi_computers`.`entities_id` IN (0, 260, 133, 136)) ) GROUP BY `glpi_computers`.`id` ORDER BY ITEM_Computer_1 ASC LIMIT 0, 50"`

(At line 2, the real username was replaced by "my_user")

**Page(s) URL**

/computer.php

**To reproduce**

1. Go to 'Assets-->Computers'
2. Click on the adjustable wrench icon --> personnal view
3. Select and add the field "external link" to the list of fields to be shown
5. Go back to personnal setting and delete the personnal view for computer view to recover the ability to search for computers

**Expected behavior**

List of computers should be shown with the field "external link" (worked in version 9.4.X)

**Logs**

```
paste logs here
Find them in *-error.log files under glpi/files/_log/

`[2020-07-24 09:42:46] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/inc/dbmysql.class.php line 306
  *** MySQL query error:
  SQL: SELECT DISTINCT `glpi_computers`.`id` AS id, 'my_user' AS currentuser,
                        `glpi_computers`.`entities_id`, `glpi_computers`.`is_recursive`,  `glpi_computers`.`name` AS `ITEM_Computer_1`,
                        `glpi_computers`.`id` AS `ITEM_Computer_1_id`,
                        `glpi_entities`.`completename` AS `ITEM_Computer_80`,  `glpi_computers`.`serial` AS `ITEM_Computer_5`,  `glpi_states`.`completename` AS `ITEM_Computer_31`,  `glpi_computertypes`.`name` AS `ITEM_Computer_4`,  `glp$
                                               '$#$',`glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`id`) ORDER BY `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`id` SEPARATOR '$$##$$')
                              AS `ITEM_Computer_126`,

                  `glpi_plugin_fusioninventory_inventorycomputercomputers`.`last_fusioninventory_update` AS `ITEM_Computer_5150`,  `glpi_plugin_fusioninventory_inventorycomputercomputers`.`last_boot` AS `ITEM_Computer_5182`,   IFNULL(GR$
                                                                         '__NULL__'),
                                                   '$#$', `glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$')
                                    AS `ITEM_Computer_145_id`,  IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`link`,
                                                                         '__NULL__'),
                                                   '$#$', `glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$')
                                    AS `ITEM_Computer_145_link`,  IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`name`,
                                                                         '__NULL__'),
                                                   '$#$', `glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$')
                                    AS `ITEM_Computer_145_name`,  IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`data`,
                                                                         '__NULL__'),
                                                   '$#$', `glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$')
                                    AS `ITEM_Computer_145_data`,  IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`open_window`,
                                                                         '__NULL__'),
                                                   '$#$', `glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`id`) SEPARATOR '$$##$$'), '__NULL__$#$')
                                    AS `ITEM_Computer_145_open_window` FROM `glpi_computers`LEFT JOIN `glpi_entities`
                                          ON (`glpi_computers`.`entities_id` = `glpi_entities`.`id`
                                              )LEFT JOIN `glpi_states`
                                          ON (`glpi_computers`.`states_id` = `glpi_states`.`id`
                                              )LEFT JOIN `glpi_computertypes`
                                          ON (`glpi_computers`.`computertypes_id` = `glpi_computertypes`.`id`
                                              )LEFT JOIN `glpi_computermodels`
                                          ON (`glpi_computers`.`computermodels_id` = `glpi_computermodels`.`id`
                                              )LEFT JOIN `glpi_manufacturers`
                                          ON (`glpi_computers`.`manufacturers_id` = `glpi_manufacturers`.`id`
                                              ) LEFT JOIN `glpi_items_operatingsystems`
                                          ON (`glpi_computers`.`id` = `glpi_items_operatingsystems`.`items_id`
                                              AND `glpi_items_operatingsystems`.`itemtype` = 'Computer'
                                              ) LEFT JOIN `glpi_operatingsystemversions`  AS `glpi_operatingsystemversions_9719987b154aaf3b42c3db32aef59090`
                                          ON (`glpi_items_operatingsystems`.`operatingsystemversions_id` = `glpi_operatingsystemversions_9719987b154aaf3b42c3db32aef59090`.`id`
                                              ) LEFT JOIN `glpi_ipaddresses`  AS `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`
                                          ON (`glpi_computers`.`id` = `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`mainitems_id`
                                              AND `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`mainitemtype` = 'Computer'
                                              AND `glpi_ipaddresses_0cc35feab42e5909929ff742b4834540`.`is_deleted` = 0 )  LEFT JOIN `glpi_plugin_fusioninventory_inventorycomputercomputers`
                    AS glpi_plugin_fusioninventory_inventorycomputercomputers
                    ON (`glpi_computers`.`id` = `glpi_plugin_fusioninventory_inventorycomputercomputers`.`computers_id` )  LEFT JOIN `glpi_links_itemtypes`
                                          ON (`glpi_links_itemtypes`.`itemtype` = 'Computer'
                                              ) LEFT JOIN `glpi_links`  AS `glpi_links_dca1a050c061633c1bacb88409d9cdbc`
                                          ON (`glpi_links_itemtypes`.`links_id` = `glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`id`
                                              (`glpi_links_dca1a050c061633c1bacb88409d9cdbc`.`entities_id` IN ('0', '259', '260', '161', '222', '232', '245', '257', '132', '133', '134', '135', '136', '137', '258', '141', '157', '179', '$
  Error: execute command denied to user 'glpi'@'%' for routine 'glpi_links_dca1a050c061633c1bacb88409d9cdbc.id'
  Backtrace :
  inc/search.class.php:1208
  inc/search.class.php:320                           Search::constructData()
  inc/search.class.php:95                            Search::getDatas()
  inc/search.class.php:80                            Search::showList()
  front/computer.php:39                              Search::show()
  {"user":"9@MY_SERVER_NAME"}
`
My real user name was replaced by my_user and the real server name was replaced by MY_SERVER_NAME.

```

**Screenshots**

![glpi](https://user-images.githubusercontent.com/45574511/88388914-c1a7dd80-cdb5-11ea-8195-0fcffc8fa531.JPG)

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

```

Information about system installation and configuration
--
Check if a new version is available
[code]   GLPI 9.5.1 ( => /var/www/html/glpi) Installation mode: TARBALL

Operating system: Linux BSALLGLPIPROD01 4.19.0-9-amd64 #1 SMP Debian 4.19.118-2+deb10u1 (2020-06-07) x86_64 PHP 7.3.19-1~deb10u1 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apc, apcu, bz2, 	calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, gmp, hash, iconv, imagick, imap, intl, json, ldap, 	libxml, mbstring, mysqli, mysqlnd, openssl, pcre, pdo_mysql, posix, readline, session, shmop, sockets, sodium, standard, 	sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib) Setup: max_execution_time="600" memory_limit="256M" post_max_size="100M" safe_mode="" session.save_handler="files" 	upload_max_filesize="50M"  Software: Apache/2.4.38 (Debian) (Apache/2.4.38 (Debian) Server at glpi.botanic.com Port 443) 	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.89 Safari/537.36 Server Software: Debian 10 	Server Version: 10.3.22-MariaDB-0+deb10u1 	Server SQL Mode:  	Parameters: glpi@localhost/glpi 	Host info: Localhost via UNIX socket 	 PHP version is at least 7.2.0 - Perfect! Sessions support is available - Perfect! Allocated memory > 64 Mio - Perfect! mysqli extension is installed ctype extension is installed fileinfo extension is installed json extension is installed mbstring extension is installed iconv extension is installed zlib extension is installed curl extension is installed gd extension is installed simplexml extension is installed intl extension is installed ldap extension is installed apcu extension is installed Zend OPcache extension is installed xmlrpc extension is installed CAS extension is installed exif extension is installed zip extension is installed bz2 extension is installed sodium extension is installed Database version seems correct (10.3.22) - Perfect! Access to timezone database (mysql) is not allowed. The log file has been created successfully. Write access to /var/www/html/glpi/config has been validated. Write access to /var/www/html/glpi/files has been validated. Write access to /var/www/html/glpi/files/_dumps has been validated. Write access to /var/www/html/glpi/files/_sessions has been validated. Write access to /var/www/html/glpi/files/_cron has been validated. Write access to /var/www/html/glpi/files/_graphs has been validated. Write access to /var/www/html/glpi/files/_lock has been validated. Write access to /var/www/html/glpi/files/_plugins has been validated. Write access to /var/www/html/glpi/files/_tmp has been validated. Write access to /var/www/html/glpi/files/_cache has been validated. Write access to /var/www/html/glpi/files/_rss has been validated. Write access to /var/www/html/glpi/files/_uploads has been validated. Write access to /var/www/html/glpi/files/_pictures has been validated. Write access to /var/www/html/glpi/marketplace has been validated. Web access to files directory is protected

GLPI_ROOT: /var/www/html/glpi GLPI_CONFIG_DIR: /var/www/html/glpi/config GLPI_VAR_DIR: /var/www/html/glpi/files GLPI_MARKETPLACE_DIR: /var/www/html/glpi/marketplace GLPI_USE_CSRF_CHECK: 1 GLPI_CSRF_EXPIRES: 7200 GLPI_CSRF_MAX_TOKENS: 100 GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org GLPI_INSTALL_MODE: TARBALL GLPI_NETWORK_MAIL: glpi@teclib.com GLPI_NETWORK_SERVICES: https://services.glpi-network.com GLPI_MARKETPLACE_PRERELEASES:  GLPI_USER_AGENT_EXTRA_COMMENTS:  GLPI_AJAX_DASHBOARD: 1 GLPI_CALDAV_IMPORT_STATE: 0 GLPI_DEMO_MODE: 0 GLPI_FORCE_EMPTY_SQL_MODE: 1 GLPI_DOC_DIR: /var/www/html/glpi/files GLPI_CACHE_DIR: /var/www/html/glpi/files/_cache GLPI_CRON_DIR: /var/www/html/glpi/files/_cron GLPI_DUMP_DIR: /var/www/html/glpi/files/_dumps GLPI_GRAPH_DIR: /var/www/html/glpi/files/_graphs GLPI_LOCAL_I18N_DIR: /var/www/html/glpi/files/_locales GLPI_LOCK_DIR: /var/www/html/glpi/files/_lock GLPI_LOG_DIR: /var/www/html/glpi/files/_log GLPI_PICTURE_DIR: /var/www/html/glpi/files/_pictures GLPI_PLUGIN_DOC_DIR: /var/www/html/glpi/files/_plugins GLPI_RSS_DIR: /var/www/html/glpi/files/_rss GLPI_SESSION_DIR: /var/www/html/glpi/files/_sessions GLPI_TMP_DIR: /var/www/html/glpi/files/_tmp GLPI_UPLOAD_DIR: /var/www/html/glpi/files/_uploads GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/ GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/ GLPI_I18N_DIR: /var/www/html/glpi/locales GLPI_VERSION: 9.5.1 GLPI_SCHEMA_VERSION: 9.5.0 GLPI_MIN_PHP: 7.2.0 GLPI_YEAR: 2020

htmlawed/htmlawed version 1.2.5 in (/var/www/html/glpi/vendor/htmlawed/htmlawed) phpmailer/phpmailer version 6.1.6 in (/var/www/html/glpi/vendor/phpmailer/phpmailer/src) simplepie/simplepie version 1.5.5 in (/var/www/html/glpi/vendor/simplepie/simplepie/library) tecnickcom/tcpdf version 6.3.5 in (/var/www/html/glpi/vendor/tecnickcom/tcpdf) michelf/php-markdown in (/var/www/html/glpi/vendor/michelf/php-markdown/Michelf) true/punycode in (/var/www/html/glpi/vendor/true/punycode/src) iamcal/lib_autolink in (/var/www/html/glpi/vendor/iamcal/lib_autolink) sabre/dav in (/var/www/html/glpi/vendor/sabre/dav/lib/DAV) sabre/http in (/var/www/html/glpi/vendor/sabre/http/lib) sabre/uri in (/var/www/html/glpi/vendor/sabre/uri/lib) sabre/vobject in (/var/www/html/glpi/vendor/sabre/vobject/lib) laminas/laminas-cache in (/var/www/html/glpi/vendor/laminas/laminas-cache/src) laminas/laminas-i18n in (/var/www/html/glpi/vendor/laminas/laminas-i18n/src) laminas/laminas-serializer in (/var/www/html/glpi/vendor/laminas/laminas-serializer/src) monolog/monolog in (/var/www/html/glpi/vendor/monolog/monolog/src/Monolog) sebastian/diff in (/var/www/html/glpi/vendor/sebastian/diff/src) elvanto/litemoji in (/var/www/html/glpi/vendor/elvanto/litemoji/src) symfony/console in (/var/www/html/glpi/vendor/symfony/console) scssphp/scssphp in (/var/www/html/glpi/vendor/scssphp/scssphp/src) laminas/laminas-mail in (/var/www/html/glpi/vendor/laminas/laminas-mail/src/Protocol) laminas/laminas-mime in (/var/www/html/glpi/vendor/laminas/laminas-mime/src) rlanvin/php-rrule in (/var/www/html/glpi/vendor/rlanvin/php-rrule/src) blueimp/jquery-file-upload in (/var/www/html/glpi/vendor/blueimp/jquery-file-upload/server/php) ramsey/uuid in (/var/www/html/glpi/vendor/ramsey/uuid/src) psr/log in (/var/www/html/glpi/vendor/psr/log/Psr/Log) psr/simple-cache in (/var/www/html/glpi/vendor/psr/simple-cache/src) mexitek/phpcolors in (/var/www/html/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors) guzzlehttp/guzzle in (/var/www/html/glpi/vendor/guzzlehttp/guzzle/src) wapmorgan/unified-archive in (/var/www/html/glpi/vendor/wapmorgan/unified-archive/src) paragonie/sodium_compat in (/var/www/html/glpi/vendor/paragonie/sodium_compat/src) phpCas version 1.3.6 in (/usr/share/php/CAS/source)

Server: 'poseidon.botanic.com', Port: '389', BaseDN: 'DC=botanic,DC=com', Connection filter: 		'(&(objectClass=user)(objectCategory=person)(!(userAccountControl:1.2.840.113556.1.4.803:=2)))', RootDN: 'CN=Systeme 		Robot,OU=Systeme,OU=Informatique,OU=Services Siège,DC=botanic,DC=com', Use TLS: none

Not active

Way of sending emails: PHP

 

fields               Name: Champs supplémentaires         Version: 1.11.0     State: Enabled 	datainjection        Name: Data Injection                 Version: 2.8.0      State: Enabled 	fusioninventory      Name: FusionInventory                Version: 9.5.0+1.0  State: Enabled 	genericobject        Name: Gestion d'objets               Version: 2.9.0      State: Enabled
[/code]


```

**Additional context**

**Comment by leecheromerevilit** [comment]
Any updates regarding this issue?

**Comment by leecheromerevilit** [comment]
Workaround

https://forum.glpi-project.org/viewtopic.php?id=168611

**Comment by redisded** [comment]
> 
> 
> Workaround
> 
> https://forum.glpi-project.org/viewtopic.php?id=168611

Hello,

not the same error message, not the same problem.

**Comment by AdrienClairembault** [comment]
Duplicate of #7915, fixed in #7930.
Updating your GLPI should solve your issue.
